src/main/java/net/neoforged/neoforge/client/settings/KeyMappingLookup.java

```diff
@@ -31,13 +32,60 @@ public class KeyMappingLookup {
      * @return the list of key mappings
      */
     public List<KeyMapping> getAll(InputConstants.Key keyCode) {
+        return this.getAll(keyCode, false);
+    }
+
+    /**
+     * Returns all active keys associated with the given key code and the active
+     * modifiers and conflict context.
+     *
+     * @param keyCode   the key being pressed
+     * @param releasing if the key is being released
+     * @return the list of key mappings
+     */
+    public List<KeyMapping> getAll(InputConstants.Key keyCode, boolean releasing) {
         List<KeyMapping> matchingBindings = new ArrayList<>();
-        KeyModifier activeModifier = KeyModifier.getActiveModifier();
-        // Apply active modifier only if the pressed key is not the modifier itself
-        // Otherwise, look for key bindings without modifiers
-        if (activeModifier == KeyModifier.NONE || activeModifier.matches(keyCode) || !matchingBindings.addAll(findKeybinds(keyCode, activeModifier))) {
+        // Get a list of all active modifiers
+        List<KeyModifier> activeModifiers = KeyModifier.getActiveModifiers();
+
+        for (var modifier : activeModifiers) {
+            // If modifier matches, add other modifiers
+            if (modifier.matches(keyCode)) {
+                // Check if binding matches with another modifier
+                for (var key : activeModifiers) {
+
+                    // Skip if modifier matches
+                    if (key.matches(keyCode)) {
+                        continue;
+                    }
+
+                    // Loop through all modifier codes
+                    for (var modifierKey : key.codes()) {
```

Comment:
Could probably be named differently to reduce the association between `modifierKey` (which is derived from the nested loop) and `modifier` (from the outer loop). Perhaps `otherModifierCode`, with the `key` loop variable renamed as `otherModifier`?